raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -241,8 +243,11 @@ def get_lock_amount_after_fees(
     Fees are taken only for the outgoing channel, which is the one with
     collateral locked from this node.
     """
-    # FIXME: user proper fee calculation
-    return PaymentWithFeeAmount(lock.amount - payee_channel.fee_schedule.flat)
+    balance = get_balance(payee_channel.our_state, payee_channel.partner_state)
+    # The fee should be calculated on the payment amount without fees. But we
```

Comment:
I think it's correct that it get's calculated on the full amount. The service that is payed is the full amount, so nothing is wrong about that. It's just the PFS that has to calculate it correctly.

Reply:
So if I want to transfer 1000 tokens which are mediated over a node with a 10% mediation fee, I'll have to pay a fee of 111 tokens, so that (1000 + 111) * 10% = 111? It's not obvious to me that this is the right way to handle it.

Reply:
Yeah, but I think that's the only think that makes sense, otherwise mediators would be underpayed, especially the early ones.

So, in `A -> B -> C -> D`, `A` wants to pay `D` 1000 tokens, `B` and `C` take 10% fee. Then `B` needs to send `C` `1000 / 0.9 = 1111` tokens and `A` even more to `B`: `(1000/0.9)/0.9 = 1234` tokens.

Reply:
Since C only needs to lock 1000 tokens, it would also be reasonable to say that B has to send only 1000 * 1.1 and thus A sends 1000 * 1.1 * 1.1 = 1210.